### ISSUE TITLE
Java 17 support for JEP 411: Deprecate the Security Manager for Removal

### ISSUE THREAD

**Issue by jorsol** [spotbugs/spotbugs#1793]
Running SpotBugs 4.4.2 on Java 17 prints this warning:
```
[INFO] --- spotbugs-maven-plugin:4.4.2.2:spotbugs (spotbugs) @ cm-api ---
[INFO] Fork Value is true
     [java] WARNING: A terminally deprecated method in java.lang.System has been called
     [java] WARNING: System::setSecurityManager has been called by edu.umd.cs.findbugs.ba.jsr305.TypeQualifierValue (file:/.m2/repository/com/github/spotbugs/spotbugs/4.4.2/spotbugs-4.4.2.jar)
     [java] WARNING: Please consider reporting this to the maintainers of edu.umd.cs.findbugs.ba.jsr305.TypeQualifierValue
     [java] WARNING: System::setSecurityManager will be removed in a future release
```

Since the Security Manager will be removed in the future, it would be great to have proper support for Java 17 and up.

On Java 18-ea the message is:
```
[INFO] --- spotbugs-maven-plugin:4.4.2.2:spotbugs (spotbugs) @ cm-api ---
[INFO] Fork Value is true
     [java] The following errors occurred during analysis:
     [java]   Unable to construct type qualifier checker javax/annotation/Nonnull$Checker
     [java]     java.lang.UnsupportedOperationException: The Security Manager is deprecated and will be removed in a future release
     [java]       At java.base/java.lang.System.setSecurityManager(System.java:411)
     [java]       At edu.umd.cs.findbugs.ba.jsr305.TypeQualifierValue.<init>(TypeQualifierValue.java:157)
     [java]       At edu.umd.cs.findbugs.ba.jsr305.TypeQualifierValue.getValue(TypeQualifierValue.java:298)
     [java]       At edu.umd.cs.findbugs.ba.jsr305.TypeQualifierValue.getValue(TypeQualifierValue.java:306)
     [java]       At edu.umd.cs.findbugs.ba.npe.TypeQualifierNullnessAnnotationDatabase.<init>(TypeQualifierNullnessAnnotationDatabase.java:70)
     [java]       At edu.umd.cs.findbugs.ba.AnalysisContext.getNullnessAnnotationDatabase(AnalysisContext.java:1055)
     [java]       At edu.umd.cs.findbugs.ba.AnalysisContext.updateDatabases(AnalysisContext.java:1008)
     [java]       At edu.umd.cs.findbugs.FindBugs2.analyzeApplication(FindBugs2.java:1061)
     [java]       At edu.umd.cs.findbugs.FindBugs2.execute(FindBugs2.java:309)
     [java]       At edu.umd.cs.findbugs.FindBugs.runMain(FindBugs.java:395)
     [java]       At edu.umd.cs.findbugs.FindBugs2.main(FindBugs2.java:1231)
```

**Comment by KengoTODA** [comment]
Duplicate of #1579
